DSCResources/ProcessMitigation/ProcessMitigation.psm1

```diff
@@ -146,78 +187,418 @@ function Test-TargetResource
+                    EnableExportAddressFilter             = $mitigation.Payload.EnableExportAddressFilter
+                    EnableExportAddressFilterPlus         = $mitigation.Payload.EnableExportAddressFilterPlus
+                    EnableImportAddressFilter             = $mitigation.Payload.EnableImportAddressFilter
+                    EnableRopCallerCheck                  = $mitigation.Payload.EnableRopCallerCheck
+                    EnableRopSimExec                      = $mitigation.Payload.EnableRopSimExec
+                    EnableRopStackPivot                   = $mitigation.Payload.EnableRopStackPivot
+                    OverrideEnableExportAddressFilter     = $mitigation.Payload.OverrideEnableExportAddressFilter
+                    OverrideEnableExportAddressFilterPlus = $mitigation.Payload.OverrideEnableExportAddressFilterPlus
+                    OverrideEnableImportAddressFilter     = $mitigation.Payload.OverrideEnableImportAddressFilter
+                    OverrideEnableRopCallerCheck          = $mitigation.Payload.OverrideEnableRopCallerCheck
+                    OverrideEnableRopSimExec              = $mitigation.Payload.OverrideEnableRopSimExec
+                    OverrideEnableRopStackPivot           = $mitigation.Payload.OverrideEnableRopStackPivot
+                }
+                SEHOP = @{
+                    Audit         = $mitigation.SEHOP.Audit
+                    Enable        = $mitigation.SEHOP.Enable
+                    OverrideSEHOP = $mitigation.SEHOP.OverrideSEHOP
+                    TelemetryOnly = $mitigation.SEHOP.TelemetryOnly
+                }
+                Heap = @{
+                    OverrideHeap     = $mitigation.Heap.OverrideHeap
+                    TerminateOnError = $mitigation.Heap.TerminateOnError
+                }
+                ChildProcess = @{
+                    Audit                        = $mitigation.ChildProcess.Audit
+                    DisallowChildProcessCreation = $mitigation.ChildProcess.DisallowChildProcessCreation
+                    OverrideChildProcess         = $mitigation.ChildProcess.OverrideChildProcess
+                }
+            }
+        }
+    }
+
+    return $resultCurrentMitigations
+}
+
+function Convert-CurrentMitigations
 {
     [CmdletBinding()]
-    [OutputType([System.Object[]])]
+    [OutputType([xml])]
     param
     (
         [Parameter(Mandatory = $true)]
-        [AllowEmptyCollection()]
-        [hashtable[]]
-        $RawResult
+        [object[]]
+        $CurrentMitigations
     )
 
-    [pscustomobject]@{
-        Enable  = @(($RawResult | Where-Object -FilterScript { $PSItem.Value -eq 'ON' }).Mitigation)
-        Disable = @(($RawResult | Where-Object -FilterScript { $PSItem.Value -eq 'OFF' }).Mitigation)
-        Default = @(($RawResult | Where-Object -FilterScript { $PSItem.Value -eq 'NOTSET' }).Mitigation)
+    [array[]]$currentMitigationsConverted = @()
```

Comment:
The array type declaration shouldn't be needed here or throughout this file.

Reply:
Removed